Automattic/wp-calypso
# Domains: Change how we handle adding privacy

#### Changes proposed in this Pull Request

Since privacy is free, we pass this as an extra param
to the registration object. No need for additional subs anymore.

Soon privacy subs are going away in order to simplify the product.

Dependency: D25010-code

#### Testing instructions

- Calypso /start (signup)
- Add a domain
- End on Checkout screen
- Try toggling privacy on/off
- No products added to cart
- Checkout (you can USE_STORE_SANDBOX to not register it really)
- Go to DARC and check the register call in the API Logs, and make sure the privacy flag is set properly

-----------------------------------------

- Calypso -> Domains -> Add
- Add a domain
- End on Checkout screen
- Try toggling privacy on/off
- No products added to cart
- Checkout (you can USE_STORE_SANDBOX to not register it really)
- Go to DARC and check the register call in the API Logs, and make sure the privacy flag is set properly
